dspback/routers/discovery.py

```diff
@@ -33,9 +33,10 @@ def is_one_char_off(str1, str2):
     return True
 
 
-async def aggregate_stages(request, stages, pageNumber = 1, pageSize = 30):
+async def aggregate_stages(request, stages, pageNumber=1, pageSize=30):
     # Insert a `$facet` to extract the total count. We specify pagination here too.
-    stages.append({ "$facet": {"docs": [{ "$skip": (pageNumber - 1) * pageSize }, { "$limit": pageSize }], "totalCount": [{ "$count": 'count'}]}})
+    stages.append({"$facet": {"docs": [{"$skip": (pageNumber - 1) * pageSize},
+                                       {"$limit": pageSize}], "totalCount": [{"$count": 'count'}]}})
 
     aggregation = await request.app.db[get_settings().mongo_database]["discovery"].aggregate(stages).to_list(pageSize)
```

Comment:
Since we are doing the limit of pageSize in the facet stage, we should pass `None `to the `to_list()` here. I tried using `to_list(None) `and it seems to work. I didn't make the change thinking I may not have done all the tests necessary. You can try making this change.